### PR TITLE
Supporting the decoding of value (inline) classes

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
- Added corresponding logic
- Added tests

### CODE REVIEW

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt [hunk 1]
```diff
@@ -173,51 +174,76 @@ public class TomlMainDecoder(
...
-                is TomlKeyValuePrimitive, is TomlStubEmptyNode -> TomlMainDecoder(nextProcessingNode, config)
-                is TomlTablePrimitive -> {
-                    val firstTableChild = nextProcessingNode.getFirstChild() ?: throw InternalDecodingException(
-                        "Decoding process failed due to invalid structure of parsed AST tree: missing children" +
-                                " in a table <${nextProcessingNode.fullTableName}>"
-                    )
-                    checkMissingRequiredProperties(firstTableChild.getNeighbourNodes(), descriptor)
-                    TomlMainDecoder(firstTableChild, config)
+    /**
+     * Entry Point into the logic, core logic of the strcuture traversal and linking the data in TOML AST
```
Comment (bishiboosh) [inline ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt:191]: small typo right here

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt [hunk 2]
```diff
@@ -173,51 +174,76 @@ public class TomlMainDecoder(
...
+     * to the descriptor and vica-versa. Basically this logic is used to iterate through data structures and do processing.
+     */
+    @Suppress("TOO_LONG_FUNCTION")
+    private fun iterateOverStructure(descriptor: SerialDescriptor, inlineFunc: Boolean): TomlAbstractDecoder =
+            when (rootNode) {
```
Comment (bishiboosh) [inline ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt:196]: as there's only two branches to the when now, could it be turned into a if ?

**Review by bishiboosh — COMMENTED** [review]
Some really small nitpicks, otherwise LGTM

**Review by bishiboosh — APPROVED** [review]

File: ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt (hunk 1 above)
Comment (akuleshov7) [reply ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMainDecoder.kt]: fixed

**Pull request merged**